geomstats/geometry/siegel.py

```diff
@@ -0,0 +1,614 @@
+        aux_4 = identity - aux_2
+        factor_1 = HermitianMatrices.powerm(aux_3, 1 / 2)
+        factor_3 = HermitianMatrices.powerm(aux_4, 1 / 2)
+        prod_1 = gs.einsum("...ij,...jk->...ik", factor_1, tangent_vec)
+        tangent_vec_at_base_point = gs.einsum("...ij,...jk->...ik", prod_1, factor_3)
+        return tangent_vec_at_base_point
+
+    def log(self, point, base_point):
+        """Compute the Siegel logarithm map.
+
+        Compute the Riemannian logarithm at point base_point
+        of point wrt the metric defined in inner_product.
+        Return a tangent vector at point base_point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Point on the Siegel space.
+        base_point : array-like, shape=[..., n, n]
+            Point on the Siegel space.
+
+        Returns
+        -------
+        log : array-like, shape=[..., n, n]
+            Riemannian logarithm at the base point.
+        """
+        point_at_zero = self.isometry(point=point, point_to_zero=base_point)
+
+        logarithm_at_zero = self.log_at_zero(point_at_zero)
+
+        log = self.tangent_vec_from_zero_to_base_point(
+            tangent_vec=logarithm_at_zero, base_point=base_point
+        )
+
+        return log
+
+    def squared_dist(self, point_a, point_b):
+        """Compute the Siegel squared distance.
+
+        Compute the Riemannian squared distance between point_a and point_b.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., n, n]
+            Point.
+        point_b : array-like, shape=[..., n, n]
+            Point.
+
+        Returns
+        -------
+        squared_dist : array-like, shape=[...]
+            Riemannian squared distance.
+
+        References
+        ----------
+        The Kahler mean of Block-Toeplitz matrices
+        with Toeplitz structured blocks
+        B. Jeuris and R. Vandebril
+        2016
+        https://epubs.siam.org/doi/pdf/10.1137/15M102112X
+        3.2. A second generalized transformation.
+        """
+        if len(point_a.shape) > len(point_b.shape):
+            point_a, point_b = point_b, point_a
+
+        data_type = (point_a + point_b).dtype
+        identity = create_identity_mat(point_b.shape, dtype=data_type)
+
+        point_a_transconj = ComplexMatrices.transconjugate(point_a)
+        point_b_transconj = ComplexMatrices.transconjugate(point_b)
+
+        factor_1 = point_b - point_a
+
+        factor_3 = ComplexMatrices.transconjugate(factor_1)
+
+        aux_2 = gs.einsum("...ij,...jk->...ik", point_a_transconj, point_b)
+
+        aux_3 = gs.einsum("...ij,...jk->...ik", point_a, point_b_transconj)
+
+        aux_4 = identity - aux_2
+
+        aux_5 = identity - aux_3
+
+        factor_2 = gs.linalg.inv(aux_4)
+
+        factor_4 = gs.linalg.inv(aux_5)
+
+        prod = gs.einsum(
+            "...ij,...jk,...kl,...lm->...im", factor_1, factor_2, factor_3, factor_4
+        )
```

Comment:
a bunch of matmuls. but probably this is faster. can be kept.

Reply:
Ok, I will keep it like this.